### PR TITLE
Add Microsoft.CodeAnalysis.EditorFeatures.dll to the MEF composition.

### PR DESCRIPTION
**Pull request by KirillOsenkov**

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
It's still broken, composition crashes, I'm investigating.

**Comment by KirillOsenkov** [comment]
It works on Windows, but on a Mac Mono is much more strict about missing assemblies, and so the lack of a few assemblies is a deal breaker.

We're missing:
Microsoft.VisualStudio.RemoteControl.dll
Microsoft.VisualStudio.CodingConventions.dll
Microsoft.VisualStudio.ImageCatalog
Microsoft.VisualStudio.Imaging

**Comment by KirillOsenkov** [comment]
FYI @jasonmalinowski

**Comment by KirillOsenkov** [comment]
Blockers listed in this Roslyn bug:
https://github.com/dotnet/roslyn/issues/22959

**Comment by KirillOsenkov** [comment]
FYI @minestarks

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/InlineRenameService.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Text;
+
+namespace MonoDevelop.Ide.Composition
+{
+	[Export(typeof(IInlineRenameService))]
+	internal class InlineRenameService : IInlineRenameService
```
Comment (jasonmalinowski) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/InlineRenameService.cs]: This one should have been fixed with the newer packages. Are you still on the older ones?

**Comment by Therzok** [comment]
Cc @slluis @mkrueger With this, we can replace our refactorings/code fixes with the Roslyn CodeFixService/CodeRefactoringService implementations.

**Comment by slluis** [comment]
Don't we have to implement IInlineRenameService and ISignatureHelpBroker first?

**Comment by KirillOsenkov** [comment]
@slluis no, these are very short-term hacks and will be deleted. These types will be moved out of EditorFeatures.dll to their EditorFeatures.Wpf.dll so we won't need to provide these anymore. The stubs are fine for now just to make MEF composition happy. As long as we don't use Roslyn's inline rename and Signature Help, these will never get called.

**Comment by KirillOsenkov** [comment]
This project is now tracking the remaining work on EditorFeatures refactoring on the Roslyn side:
https://github.com/dotnet/roslyn/projects/32

**Comment by KirillOsenkov** [comment]
@slluis @manish it would be nice for this to land in master to unblock a lot of work. What extra validation do we want here? 

/cc @DavidKarlas @Therzok

**Pull request merged**